radsat-sk/operation/message/protobuf/types/RFileTransfer.pb.h

```diff
@@ -9,87 +9,348 @@
+
+#define ImagePacket_FIELDLIST(X, a) \
+X(a, STATIC,   SINGULAR, UINT32,   id,                1) \
+X(a, STATIC,   SINGULAR, UENUM,    type,              2) \
 X(a, STATIC,   SINGULAR, BYTES,    data,              3)
-#define FileTransferPacket_CALLBACK NULL
-#define FileTransferPacket_DEFAULT NULL
+#define ImagePacket_CALLBACK NULL
+#define ImagePacket_DEFAULT NULL
 
 extern const pb_msgdesc_t FileTransferMessage_msg;
-extern const pb_msgdesc_t FileTransferResponse_msg;
-extern const pb_msgdesc_t FileTransferPacket_msg;
+extern const pb_msgdesc_t ObcTelemetry_msg;
+extern const pb_msgdesc_t TransceiverTelemetry_msg;
+extern const pb_msgdesc_t TransceiverTelemetry_receiverTelemetry_msg;
+extern const pb_msgdesc_t TransceiverTelemetry_transmitterTelemetry_msg;
+extern const pb_msgdesc_t CameraTelemetry_msg;
+extern const pb_msgdesc_t EpsTelemetry_msg;
+extern const pb_msgdesc_t BatteryTelemetry_msg;
+extern const pb_msgdesc_t AntennaTelemetry_msg;
+extern const pb_msgdesc_t DosimeterData_msg;
+extern const pb_msgdesc_t DosimeterData_DosimeterBoardData_msg;
+extern const pb_msgdesc_t ImagePacket_msg;
 
 /* Defines for backwards compatibility with code written before nanopb-0.4.0 */
 #define FileTransferMessage_fields &FileTransferMessage_msg
-#define FileTransferResponse_fields &FileTransferResponse_msg
-#define FileTransferPacket_fields &FileTransferPacket_msg
+#define ObcTelemetry_fields &ObcTelemetry_msg
+#define TransceiverTelemetry_fields &TransceiverTelemetry_msg
+#define TransceiverTelemetry_receiverTelemetry_fields &TransceiverTelemetry_receiverTelemetry_msg
+#define TransceiverTelemetry_transmitterTelemetry_fields &TransceiverTelemetry_transmitterTelemetry_msg
+#define CameraTelemetry_fields &CameraTelemetry_msg
+#define EpsTelemetry_fields &EpsTelemetry_msg
+#define BatteryTelemetry_fields &BatteryTelemetry_msg
+#define AntennaTelemetry_fields &AntennaTelemetry_msg
+#define DosimeterData_fields &DosimeterData_msg
+#define DosimeterData_DosimeterBoardData_fields &DosimeterData_DosimeterBoardData_msg
+#define ImagePacket_fields &ImagePacket_msg
 
 /* Maximum encoded size of messages (where known) */
-#define FileTransferMessage_size                 228
-#define FileTransferResponse_size                33
-#define FileTransferPacket_size                  225
+#define FileTransferMessage_size                 214
+#define ObcTelemetry_size                        24
+#define TransceiverTelemetry_size                112
+#define TransceiverTelemetry_receiverTelemetry_size 57
+#define TransceiverTelemetry_transmitterTelemetry_size 51
+#define CameraTelemetry_size                     6
+#define EpsTelemetry_size                        0
+#define BatteryTelemetry_size                    0
+#define AntennaTelemetry_size                    0
+#define DosimeterData_size                       84
+#define DosimeterData_DosimeterBoardData_size    40
+#define ImagePacket_size                         211
```

Comment:
I thought the Image Packet size is supposed to be 200 to allow more overhead

Reply:
Internally it has 200 bytes of payload data (you can see that set in `RFileTransfer.options`) but then there's also additional metadata within the `ImageData` struct (stuff like frame numbers, size, etc.)